src/utils/handleSubmit.js

```diff
@@ -0,0 +1,40 @@
+import Popup from "../components/Popup.js";
+
+export function handleSubmit(
+  submitAction,
+  popupInstance,
+  submitText = "Save",
+  loadingText = "Saving..."
+) {
+  // Check if popupInstance is an instance of Popup
+  if (!(popupInstance instanceof Popup)) {
+    console.error("The provided popupInstance is not an instance of Popup.");
+    return Promise.reject("Invalid popup instance");
+  }
+
+  const popup = popupInstance.popupElement;
+  const submitButton = popupInstance._submitButton;
+
+  if (!submitButton) {
+    console.error("Submit button not found in popup.");
+    return Promise.reject("Submit button not found");
+  }
+
+  submitButton.disabled = true;
+  submitButton.textContent = loadingText;
```

Comment:
`  submitButton.textContent = loadingText;
`

You have `renderLoading` for changing the button text

```js
  popupInstance.renderLoading(true, loadingText);

```